client/my-sites/earn/launchpad/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { CircularProgressBar } from '@automattic/components';
+import { useLaunchpad } from '@automattic/data-stores';
+import { Launchpad, Task, setUpActionsForTasks } from '@automattic/launchpad';
+import { useTranslate } from 'i18n-calypso';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSelector } from 'calypso/state';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+const EarnLaunchpad = () => {
+	const translate = useTranslate();
+
+	const site = useSelector( ( state ) => getSelectedSite( state ) );
+	const isNewsletterOrWriteIntent =
+		site?.options?.site_intent === 'newsletter' || site?.options?.site_intent === 'write';
+
+	const {
+		data: { checklist },
+	} = useLaunchpad( site?.slug ?? null, getChecklistSlug() );
+
+	const numberOfSteps = checklist?.length || 0;
+	const completedSteps = ( checklist?.filter( ( task: Task ) => task.completed ) || [] ).length;
+	const tasklistCompleted = completedSteps >= numberOfSteps;
+
+	const tracksData = {
+		recordTracksEvent,
+		checklistSlug: getChecklistSlug(),
+		tasklistCompleted,
+		launchpadContext: 'earn',
+	};
+
+	function getChecklistSlug() {
+		return isNewsletterOrWriteIntent ? 'earn-newsletter' : '';
+	}
+
+	function shouldLoadLaunchpad() {
+		return isNewsletterOrWriteIntent && ! tasklistCompleted && numberOfSteps > 0;
+	}
+
+	const taskFilter = ( tasks: Task[] ) => {
+		return setUpActionsForTasks( {
+			tasks,
+			siteSlug: site?.slug ?? null,
+			tracksData,
+		} );
+	};
+
+	if ( ! shouldLoadLaunchpad() ) {
+		return null;
+	}
+
+	return (
+		<div className="earn__launchpad">
+			<div className="earn__launchpad-header">
+				<h2 className="earn__launchpad-title">
+					{ translate( 'Create your paid newsletter in two steps.' ) }
+				</h2>
+				<div className="earn__launchpad-progress-bar-container">
+					<CircularProgressBar
+						size={ 30 }
+						enableDesktopScaling
+						numberOfSteps={ numberOfSteps }
+						currentStep={ completedSteps }
+					/>
+				</div>
+			</div>
+			<Launchpad
+				siteSlug={ site?.slug ?? null }
+				checklistSlug={ getChecklistSlug() }
+				taskFilter={ taskFilter }
+			/>
+		</div>
+	);
+};
```

Comment:
It feels somewhat redundant/unnecessary to fetch the checklist data when we can already tell that we don't want to render the launchpad. I know that we need all the hooks to run before we return from the component, but I am wondering whether it may be worth avoiding the unnecessary API calls by having a wrapper component that checks the site intent and returns null when we don't have a matching intent and have most of this logic in an inner component.
```suggestion
const EarnLaunchpadActual = () => {
	const translate = useTranslate();

	const site = useSelector( ( state ) => getSelectedSite( state ) );
	const checklistSlug = 'earn-launchpad';

	const {
		data: { checklist },
	} = useLaunchpad( site?.slug ?? null, checklistSlug );

	const numberOfSteps = checklist?.length || 0;
	const completedSteps = ( checklist?.filter( ( task: Task ) => task.completed ) || [] ).length;
	const tasklistCompleted = completedSteps >= numberOfSteps;

	const tracksData = {
		recordTracksEvent,
		checklistSlug,
		tasklistCompleted,
		launchpadContext: 'earn',
	};

	function shouldLoadLaunchpad() {
		return ! tasklistCompleted && numberOfSteps > 0;
	}

	const taskFilter = ( tasks: Task[] ) => {
		return setUpActionsForTasks( {
			tasks,
			siteSlug: site?.slug ?? null,
			tracksData,
		} );
	};

	if ( ! shouldLoadLaunchpad() ) {
		return null;
	}

	return (
		<div className="earn__launchpad">
			<div className="earn__launchpad-header">
				<h2 className="earn__launchpad-title">
					{ translate( 'Create your paid newsletter in two steps.' ) }
				</h2>
				<div className="earn__launchpad-progress-bar-container">
					<CircularProgressBar
						size={ 30 }
						enableDesktopScaling
						numberOfSteps={ numberOfSteps }
						currentStep={ completedSteps }
					/>
				</div>
			</div>
			<Launchpad
				siteSlug={ site?.slug ?? null }
				checklistSlug={ checklistSlug }
				taskFilter={ taskFilter }
			/>
		</div>
	);
};

const EarnLaunchpad = () => {
	const site = useSelector( ( state ) => getSelectedSite( state ) );
	const isNewsletterOrWriteIntent =
		site?.options?.site_intent === 'newsletter' || site?.options?.site_intent === 'write';
	
	if ( isNewsletterOrWriteIntent ) {
		return null;
	}

	return <EarnLaunchpadActual />;
};
```

Reply:
Just a follow up comment on this - I've since adjusted the PR to show the Launchpad for all site intents. So the extra tier of conditionals here for newsletter or write intent have been removed.